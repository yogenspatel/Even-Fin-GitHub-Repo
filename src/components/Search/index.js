import React from "react";
import { connect } from "react-redux";
import { SearchData, toggleGreenColor, SearchPaginatedData } from "../../actions/index";
import PropTypes from "prop-types";

class Search extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      searchInput: "",
      class: "green"
    };
  }
  updateSearchValue = e => {
    this.setState(
      {
        searchInput: e.target.value
      },
      () => {
        this.props.SearchData(this.state.searchInput, this.props.data);
        this.props.SearchPaginatedData(this.state.searchInput, this.props.data);
      }
    );
  };

  toggleGreenColor = () => {
    this.state.class ? this.setState({ class: "" }) : this.setState({ class: "green" });
    this.props.toggleGreenColor(this.state.class);
  };

  render() {
    return (
      <React.Fragment>
        <input
          name="search"
          id="search"
          placeholder="Search records"
          onChange={this.updateSearchValue}
          value={this.state.searchInput}
        />
        <button className="btn btn-info" onClick={this.toggleGreenColor}>
          Green {"<-->"} Black
        </button>
      </React.Fragment>
    );
  }
}

Search.propTypes = {
  SearchData: PropTypes.func,
  SearchPaginatedData: PropTypes.func,
  data: PropTypes.oneOfType([PropTypes.object, PropTypes.array]),
  toggleGreenColor: PropTypes.func
};

Search.defaultProps = {
  SearchData: () => {},
  SearchPaginatedData: () => {},
  data: {},
  toggleGreenColor: () => {}
};

export { Search };

export default connect(
  null,
  { SearchData, toggleGreenColor, SearchPaginatedData }
)(Search);
