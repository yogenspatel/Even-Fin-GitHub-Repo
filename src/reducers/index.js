import { createStore, combineReducers } from "redux";
import { reducer as reduxFormReducer } from "redux-form";
import searchReducer from "./search";

const rootReducer = combineReducers({
  form: reduxFormReducer, // mounted under "form"
  searchResults: searchReducer
});

function configureStore(state = {}) {
  return createStore(rootReducer, state);
}

export { configureStore };
export default rootReducer;
