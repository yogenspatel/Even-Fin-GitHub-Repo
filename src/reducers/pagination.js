/* eslint-disable no-case-declarations */
import { PAGINATED_DATA } from "../actions";

export default (state = null, action) => {
  switch (action.type) {
    case PAGINATED_DATA:
      const { payload, noOfItems, pageSize, searchData } = action;
      return { ...state, paginated_data: payload, noOfItems, pageSize, searchData };
    default:
      return state;
  }
};
