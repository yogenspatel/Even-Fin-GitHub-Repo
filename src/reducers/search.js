import { PERFORM_SEARCH } from '../actions';

export default (state = null, action) => {
  switch (action.type) {
    case PERFORM_SEARCH:
      return action.payload.data.items;
    default:
      break;
  }
  return state;
};
