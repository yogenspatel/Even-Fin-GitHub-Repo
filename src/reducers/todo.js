import { GET_TODO_DATA, SET_LOADING_STATUS, SET_DATA_ERROR, SORT_DATA } from "../actions";

export default (state = null, action) => {
  switch (action.type) {
    case SET_LOADING_STATUS:
      return { ...state, status: action.status };
    case SET_DATA_ERROR:
      return { ...state, status: action.error };
    case GET_TODO_DATA:
      return action.payload ? action.payload : state;
    case SORT_DATA:
      console.log('state: ', action.sort_data);
      return action.sort_data;
    default:
      return state;
  }
};
