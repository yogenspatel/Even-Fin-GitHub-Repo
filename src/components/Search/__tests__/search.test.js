/* eslint-disable no-undef */

import React from "react";
import ReactDOM from "react-dom";
import { Search } from "../../Search";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { configureStore } from "../../../reducers";

describe("Search Component renders correctly", () => {
  it("Search renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Search />, div);
  });

  it("Search Component with Mock Data", () => {
    const store = configureStore();
    const tree = shallow(<Search store={store} />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
