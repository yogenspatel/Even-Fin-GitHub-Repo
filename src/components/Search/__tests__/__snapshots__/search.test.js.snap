// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Component renders correctly Search Component with Mock Data 1`] = `
<Fragment>
  <input
    id="search"
    name="search"
    onChange={[Function]}
    placeholder="Search records"
    value=""
  />
  <button
    className="btn btn-info"
    onClick={[Function]}
  >
    Green 
    &lt;--&gt;
     Black
  </button>
</Fragment>
`;
